python/ray/dashboard/modules/metrics/templates.py

```diff
@@ -22,15 +24,27 @@
       path: {dashboard_output_folder}
 """
 
-GRAFANA_DATASOURCE_TEMPLATE = """apiVersion: 1
 
-datasources:
-  - name: {prometheus_name}
-    url: {prometheus_host}
-    type: prometheus
-    isDefault: true
-    access: proxy
-"""
+def GRAFANA_DATASOURCE_TEMPLATE(
+    prometheus_name, prometheus_host, jsonData, secureJsonData
+):
+    return yaml.safe_dump(
+        {
+            "apiVersion": 1,
+            "datasources": [
+                {
+                    "name": prometheus_name,
+                    "url": prometheus_host,
+                    "type": "prometheus",
+                    "isDefault": True,
+                    "access": "proxy",
+                    "jsonData": jsonData,
+                    "secureJsonData": secureJsonData,
+                }
+            ],
+        }
+    )
```

Comment:
Since a simple string template is now not enough for complex values in `jsonData` and `secureJsonData` and there could be some values that need to be escaped to fit into yaml, I think using `yaml.safe_dump` is better.